src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,279 @@
+const Papa = require('papaparse')
+const fs = require('fs')
+
+const ApiHandler = require('../ApiHandler.class')
+
+// Buffer which handle error during convertion
+const convertionErrorBuffer = {
+  error: [],
+  reset() {
+    this.error = []
+  },
+}
+
+const ERROR_PRINT_SIZE = 5
+/**
+ * Class CsvToHttp - convert a csv file into http request such as POST/PUT/PACTH
+ */
+class CsvToHttp extends ApiHandler {
+  /**
+   * Constructor for CsvToHttp
+   * @constructor
+   * @param {Object} applicationParameters - The application parameters
+   * @param {Engine} engine - The Engine
+   * @return {void}
+   */
+  constructor(applicationParameters, engine) {
+    super(applicationParameters, engine)
+
+    const {
+      applicativeHostUrl, requestMethod, bodyMaxLength, acceptUnconvertedRows,
+      authentication, csvDelimiter, mapping, proxy = null,
+    } = applicationParameters.CsvToHttp
+
+    // Creation of an entity request with is composed of all necessaries informations
+    this.request = {
+      host: applicativeHostUrl,
+      method: requestMethod,
+      authenticationField: authentication,
+    }
+
+    this.bodyMaxLength = bodyMaxLength
+    this.acceptUnconvertedRows = acceptUnconvertedRows
+
+    this.mapping = mapping || {}
+    this.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.canHandleFiles = true
+    this.canHandleValues = false
+  }
+
+  /**
+   * Send the file.
+   * @param {String} filePath - The path of the file
+   * @return {Promise} - The send status
+   */
+  async handleFile(filePath) {
+    // Verify that the file receive is a csv one
+    const regexExp = RegExp('.csv$')
+
+    if (!regexExp.test(filePath)) {
+      this.logger.error(`Invalid file format (.csv file expected), file (${filePath} skipped`)
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    const csvDataParsed = await this.parseCsvFile(filePath)
+
+    if (csvDataParsed.length === 0) {
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    // The csv parsing is a success so we begin to map the content
+    // Initialize the body to an empty array
+    const httpBody = this.convertToHttpBody(csvDataParsed)
+
+    if (httpBody.length !== 0) {
+      if (!await this.sendData(httpBody)) {
+        this.logger.error('Impossible to send data')
+        return ApiHandler.STATUS.LOGIC_ERROR
+      }
+    }
+
+    // Logs all the erros
+    if (convertionErrorBuffer.error.length > 0) {
+      this.logger.error(`${convertionErrorBuffer.error.length} convertions error`)
+
+      if (convertionErrorBuffer.error.length > ERROR_PRINT_SIZE) {
+        for (let i = 0; i < ERROR_PRINT_SIZE; i += 1) {
+          this.logger.error(`${convertionErrorBuffer.error[i]}`)
+        }
+      } else {
+        convertionErrorBuffer.error.forEach((index) => {
+          this.logger.error(`${convertionErrorBuffer.error[index]}`)
```

Comment:
This will not work : forEach first parameter is the entity, second parameter is the index. Here you name the entity index which is wrong and use it as an index, which is wrong too. It must be written like this : 
       convertionErrorBuffer.error.forEach((error) => {
          this.logger.error(`${error}`)
        })
        
A test may be usefule to check if the logger is called with the expected error